src/Calypso-SystemTools-FullBrowser/ClyFullBrowserMorph.class.st

```diff
@@ -714,6 +712,19 @@ ClyFullBrowserMorph >> showAllMethods [
 	methodView showQuery: methodQuery
 ]
 
+{ #category : 'navigation' }
+ClyFullBrowserMorph >> showInPackageView: aPackage [
+
+	"If true we are in the project view and we need to find the package and expand it"
+	(packageView showsItemsFromQuery: ClyAllProjectsQuery)
+		ifTrue: [
+			packageView dataSource allElements
+				detect: [ :element | (element actualObject isKindOf: Project) and: [ element actualObject packages includes: aPackage ] ]
+				ifFound: [ :element | packageView dataSource expand: element ]
+				ifNone: [ "this should not happen but who knows with calypso.." self switchToPackages ] ]
+		ifFalse: [ self switchToPackages ]
```

Comment:
There is API to lookup items and it can be rewritten like: 
```Smalltalk
showInPackageView: aPackage

	(packageView showsItemsFromQuery: ClyAllProjectsQuery)
		ifTrue: [
			self projectSelection selectItemsWhere: [ :each | 
				(each representsItemOfType: Project) and: [ each actualObject includesPackage: aPackage ]].
			self projectSelection items first expand ].
	self packageSelection selectItemsWith: { aPackage }
```

Reply:
I'll probably do another pass soon and integrate your feedback! Thanks